SecurityPkg/Tcg/Tcg2Pei/Tcg2Pei.c

```diff
@@ -1068,11 +1127,18 @@ PeimEntryMP (
   Status = PeiServicesInstallPpi (&mTcgPpiList);
   ASSERT_EFI_ERROR (Status);
 
-  // MU_CHANGE_103691
-  // MU_CHANGE [BEGIN] - Add support for measurements extended before Tcg2 stack is available.
+  // MU_CHANGE [BEGIN] - 103691 - Add support for measurements extended before Tcg2 stack is available.
   CreateTcg2PreUefiEventLogEntries ();
   // MU_CHANGE [END]
 
+  // MU_CHANGE [BEGIN] - Measure Insecure Device State into PCR07 and PCR04
+  if (GetDeviceState() & GetInsecuteDeviceStateSetting()) {
+    Status = MeasureInsecureDeviceState ();
+    // TODO we don't want to do anything if the measurement fails? Just override the status with the next measurement?
+    // E.g. could fail here but succeed the last measurement and return success. 

```

Comment:
I would at minimum assert and consider panicking, based on the conversation above.